### ISSUE TITLE
Sidebar : Combination of role 'complementary' and aria-modal attribute causes accessibility violation.

### ISSUE THREAD

**Issue by rohitpant-wtw** [primefaces/primeng#14818]
### Describe the bug

Currently the role for the prime ng sidebar is set as 'complementary' (not configurable but hardcoded in library [Pic-1]) which is correct for a sidebar panel but due to aria-modal attribute being present in the element we are getting accessibility violation (Pic-2).

As per the aria-modal description, it is only recommended for role values like 'dialog' and 'alertdialog' (https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal)

We would ideally want the sidebar to have role 'complementary' with overlay mask behind the sidebar and the aria-modal attribute not present at all (even a value 'false' will result in accessibility violation) in the HTML element.

<div role="complementary" class="ng-trigger ng-trigger-panelState sidebar p-sidebar p-sidebar-active p-sidebar-left ng-star-inserted" ng-reflect-ng-class="[object Object]" ng-reflect-ng-style="[object Object]" data-pc-name="sidebar" data-pc-section="root" style="width: 400px; z-index: 1102;">

Otherwise if you are planning to make the [attr.role] as configurable then we will also need to give ability to add [attr.aria-label] for avoiding further accessibility violations.

[Pic-1]
![image](https://github.com/primefaces/primeng/assets/152407667/1aaa021e-39d5-4d01-a237-25b417bdc14b)

[Pic-2]
![image](https://github.com/primefaces/primeng/assets/152407667/875af6ad-7b73-4995-a8a5-04257e5b6c2c)

### Environment

OS: Windows 10

### Angular version

16.2.10

### PrimeNG version

16.5.1

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.16.0

### Browser(s)

Chrome, Edge

### Steps to reproduce the behavior

1. Open primeng sidebar page (https://primeng.org/sidebar) on Chrome/Edge browser.
2. Open devtools (F12).
3. Open Lighthouse and select accessibility from categories (Pic-1)
3. Run 'Analyze page load' and open the sidebar on the primeng website when the test is running (Pic-2).
4. Once the report is ready accessibility violation is present (Pic-3).

[Pic-1]
![image](https://github.com/primefaces/primeng/assets/152407667/c84b2e24-a00f-4426-8b40-dcc76067c4bc)

[Pic-2]
![image](https://github.com/primefaces/primeng/assets/152407667/1062cac3-75f6-47ab-b0fc-0123c99f7aad)

[Pic-3]
![image](https://github.com/primefaces/primeng/assets/152407667/f430f407-4da7-4d2d-b906-5985eeca91d9)

### Expected behavior

1. Sidebar should have role  'complementary'.   --> Already present
2. Sidebar should have overlay mask behind the sidebar. --> Already present
3. Sidebar div element should not have aria-modal attribute as the combination of role 'complementary' and attribute 'aria-modal' (either true/false) is causing accessibility violation 

<div role="complementary" class="ng-trigger ng-trigger-panelState sidebar p-sidebar p-sidebar-active p-sidebar-left ng-star-inserted" data-pc-name="sidebar" data-pc-section="root">

**Comment by cetincakiroglu** [comment]
Hi,

Thanks a lot for the detailed explanation, we'll address this issue with the 17.8.0 release.
